test/apps/network/tcp_err.c

```diff
@@ -478,3 +478,73 @@ FN_TEST(self_connect)
 	TEST_SUCC(close(sk));
 }
 END_TEST()
+
+FN_TEST(listen_at_the_same_address)
+{
+	int sk_listen1;
+	int sk_listen2;
+
+	sk_listen1 = TEST_SUCC(socket(PF_INET, SOCK_STREAM, 0));
+	sk_listen2 = TEST_SUCC(socket(PF_INET, SOCK_STREAM, 0));
+
+	int reuse_option = 1;
+	CHECK(setsockopt(sk_listen1, SOL_SOCKET, SO_REUSEADDR, &reuse_option,
+			 sizeof(reuse_option)));
+	CHECK(setsockopt(sk_listen2, SOL_SOCKET, SO_REUSEADDR, &reuse_option,
```

Comment:
`TEST_SUCC` to be consistent with `bind_and_connect_same_address` and a bit more robust to errors.